src/java.desktop/share/classes/java/awt/image/Kernel.java

```diff
@@ -59,21 +59,38 @@ public class Kernel implements Cloneable {
      * @param width         width of the kernel
      * @param height        height of the kernel
      * @param data          kernel data in row major order
+     * @throws IllegalArgumentException if {@code data} is null
```

Comment:
Is there any particular reason why IllegalArgumentException was chosen instead of NullPointerException?

Reply:
>Current spec already throws IAE in one of the validation scenario, so throwing the same IAE for other input argument validation is uniform.

We usually throw an NPE when the value is null, since you can’t validate any properties on a null reference, and it’s more specific than an IAE. Moreover, NPE was thrown before, so this won’t change the behavior.

Reply:
NPE which was thrown before was undefined behaviour.
There is no objection in CSR request also. I would like to continue keeping usage of IAE.

If majority of reviewers have objections, we can take it up in follow-up issue.